### ISSUE TITLE
Sometimes active trades list are not updated after position is opened, updated or closed

### ISSUE THREAD

**Issue by kayalight** [DistributedCollective/Sovryn-frontend#374]
the trading activity should update automatically, it would be great to even show a pending trade the sec the user closes the pop up

please contact me and discuss this with me

**Comment by creed-victor** [comment]
> the trading activity should update automatically
Speculations of why data may not be updated automatically:
- frontend doesn't receive last block information from node
- frontend block watcher may disconnect if window is left without activity for too long
- frontend disconnected from node because of internet connection and failed to reconnect to block watcher
- node itself dropped connection for frontend and frontend failed to reconnect to block watcher

> it would be great to even show a pending trade the sec the user closes the pop up
This part should be ignored in this task and new task as feature request should be created to discuss it

**Comment by soulBit** [comment]
Tested this on latest swap page - pending transaction appears (and disappears once confirmed) correctly, but the confirmed tx doesn't show in swap history till page is refreshed - we should check that swap history is being re-fetched periodically, or if this is already happening then we should make sure the tx is not being de-duplicated because of pending tx

**Comment by tiltom** [comment]
I looked at Swap page and the reason why the history table is not automatically refreshed is pretty simple - we fetch the history transactions in an `useEffect` that is fired only on `account` change (= only after refresh).

The solution, however, is not so simple. I added a callback after successful transactions and changed the `key` on `onGoingTransactions` because of the possible clash. This should be sufficient but I found out that we don't have the data immediately after a transaction is successful, so the history table data fetch returns the same array of items. The problem is, our backend endpoint (`${url}/events/conversion-swap/${account}`) doesn't have the data right away. When I tried it to call the `callback` with `setTimeout`, it returned the correct items but it's not a good solution for multiple reasons:

- We don't know how much delay we need so this is just an estimate.
- We would show our users a table without the newest items and then it would suddenly refresh (it feels buggy).
- Usage of `setTimeout` is not a good practice.

Any suggestions @creed-victor or @pietro-maximoff ?

**Comment by grinry** [comment]
It's possible to use `useBlockSync` - it returns block number dapp detected that `account` had confirmed transaction.
Ofc it will not help because we don't have data in the backend immediately - maybe it would make sense to have interval only for the next 5 minutes after useBlockSync changed.

**Comment by tiltom** [comment]
> It's possible to use `useBlockSync` - it returns block number dapp detected that `account` had confirmed transaction.
> Ofc it will not help because we don't have data in the backend immediately - maybe it would make sense to have interval only for the next 5 minutes after useBlockSync changed.

Thanks for the idea, I will take a look at it :)
